drink/src/chain_api.rs

```diff
@@ -1,158 +1,117 @@
-    ) -> DispatchResultWithInfo<<RuntimeCall<R> as Dispatchable>::PostInfo>;
-
-    /// Return the events of the current block so far.
-    fn get_current_block_events(&mut self) -> Vec<EventRecordOf<R>>;
-
-    /// Reset the events of the current block.
-    fn reset_current_block_events(&mut self);
-}
-
-impl<R: Runtime> ChainApi<R> for Sandbox<R> {
-    fn current_height(&mut self) -> BlockNumberFor<R> {
-        self.externalities
-            .execute_with(|| frame_system::Pallet::<R>::block_number())
-    }
-
-    fn build_block(&mut self) -> DrinkResult<BlockNumberFor<R>> {
-        let mut current_block = self.current_height();
-        self.externalities.execute_with(|| {
-            let block_hash = R::finalize_block(current_block).map_err(Error::BlockFinalize)?;
-            current_block.saturating_inc();
-            R::initialize_block(current_block, block_hash).map_err(Error::BlockInitialize)?;
-            Ok(current_block)
-        })
-    }
-
-    fn add_tokens(&mut self, address: AccountIdFor<R>, amount: u128) {
-        self.externalities.execute_with(|| {
-            let _ = pallet_balances::Pallet::<R>::deposit_creating(&address, amount);
-        });
-    }
-
-    fn balance(&mut self, address: &AccountIdFor<R>) -> u128 {
-        self.externalities
-            .execute_with(|| pallet_balances::Pallet::<R>::free_balance(address))
-    }
-
-    fn get_timestamp(&mut self) -> <R as pallet_timestamp::Config>::Moment {
-        self.externalities
-            .execute_with(|| pallet_timestamp::Pallet::<R>::get())
-    }
-
-    fn set_timestamp(&mut self, timestamp: <R as pallet_timestamp::Config>::Moment) {
-        self.externalities
-            .execute_with(|| pallet_timestamp::Pallet::<R>::set_timestamp(timestamp));
+    ) -> DispatchResultWithInfo<<RuntimeCall<R> as Dispatchable>::PostInfo> {
+        self.execute_with(|| call.dispatch(origin))
     }
+}
 
-    fn dry_run<T>(&mut self, action: impl FnOnce(&mut Self) -> T) -> T {
-        // Make a backup of the backend.
-        let backend_backup = self.externalities.as_backend();
-
-        // Run the action, potentially modifying storage. Ensure, that there are no pending changes
-        // that would affect the reverted backend.
-        let result = action(self);
-        self.externalities
-            .commit_all()
-            .expect("Failed to commit changes");
-
-        // Restore the backend.
-        self.externalities.backend = backend_backup;
-
-        result
+impl<R: pallet_balances::Config> Sandbox<R> {
+    /// Add tokens to an account.
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - The address of the account to add tokens to.
+    /// * `amount` - The number of tokens to add.
+    pub fn add_tokens(
+        &mut self,
+        address: AccountIdFor<R>,
+        amount: BalanceOf<R>,
+    ) -> Result<BalanceOf<R>, DispatchError> {
+        self.execute_with(|| pallet_balances::Pallet::<R>::mint_into(&address, amount))
```

Comment:
ideally we keep existing pallet api names, but I did not want to break current API, if that's ok to make breaking change here and other apis, I am happy to do so as part of this PR

Reply:
we can break it, no problem with that (we will just have to bump version appropriately)

Reply:
bumped to 0.7